os/arch/arm/src/armv7-a/arm_cpupause.c

```diff
@@ -417,4 +417,53 @@ int up_cpu_resume(int cpu)
 	return OK;
 }
 
+/****************************************************************************
+ * Name: up_cpu_pause_all
+ *
+ * Description:
+ *   pause all the CPUs other than the current cpu. Internally, it calls 
+ * 	 up_cpu_pause() api to stop all the CPU's.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int up_cpu_pause_all()
+{
+	int me = sched_getcpu();
+	for (int cpu = 0; cpu < CONFIG_SMP_NCPUS; cpu++) {
+		if (cpu != me) {
+			/* Pause the CPU */
+			up_cpu_pause(cpu);
+			/* Wait while the pause request is pending */
+			while (up_cpu_pausereq(cpu)) {
+			}
+		}
+	}
+	return 0;
+}
+
+/****************************************************************************
+ * Name: up_cpu_resume_all
+ *
+ * Description:
+ *   Resume all the CPUs which were paused earlier.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int up_cpu_resume_all()
+{
+	int me = sched_getcpu();
+	for (int cpu = 0; cpu < CONFIG_SMP_NCPUS; cpu++) {
+		if (cpu != me) {
+			/* Resume the CPU */
+			up_cpu_resume(cpu);
```

Comment:
Don't you need to check resume here like pause all does?

Reply:
yes, it's better to wait till the cpu gets resume. Let me add it.